packages/TrilinosInstallTests/find_package_Trilinos/CMakeLists.txt

```diff
@@ -1,4 +1,7 @@
-cmake_minimum_required(VERSION 3.17)
+cmake_minimum_required(VERSION 3.0)
```

Comment:
I thought our required cmake version was going to be bumped from 3.17 to 3.23. Why set this to 3.0? Have seen this in a few other places as well.

Reply:
> I thought our required cmake version was going to be bumped from 3.17 to 3.23. Why set this to 3.0? Have seen this in a few other places as well.

@rppawlo, yes, but we don't impose that on downstream CMake projects that call `cmake_minimim_required(VERSION <cmake-version>)` with `<cmake-version>` lower than even 3.3 (which is what happened with [TRILINOSHD-128](https://sems-atlassian-son.sandia.gov/jira/servicedesk/customer/portal/7/TRILINOSHD-128)).  These files are meant to act like external CMake projects that my use a lower version of Trilinos (even lower than 3.3) so that we can validate that we are doing our part to support such projects (because these projects actually exist).

The idea with CMake is that your project may require a higher version of CMake than the `<Package>Config.cmake` files produced by your project so you can support projects that require an older version of CMake.  It seems that the goal is to write  `<Package>Config.cmake` files in order to require the lowest version of CMake possible by downstream customers.  In fact, the package-config targets files generated by CMake itself by the command [`export(EXPORT ... FILE <filename>)`](https://cmake.org/cmake/help/v3.17/command/export.html) only requires CMake version 3.0.0 and uses polices from CMake 2.6.0!  You can see this in the `<Package>Targets.cmake` files generated by CMake with Trilinos with the top showing:

```
# Generated by CMake

if("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" LESS 2.5)
   message(FATAL_ERROR "CMake >= 2.6.0 required")
endif()
cmake_policy(PUSH)
cmake_policy(VERSION 2.6)
#----------------------------------------------------------------
# Generated CMake target import file.
#----------------------------------------------------------------

# Commands may need to know the format version.
set(CMAKE_IMPORT_FILE_VERSION 1)

if(CMAKE_VERSION VERSION_LESS 3.0.0)
  message(FATAL_ERROR "This file relies on consumers using CMake 3.0.0 or greater.")
endif()
```

(Take a look at the `<Package>Targets.cmake` files generated under `<buildDir>/cmake_packages/` to see for yourself.)

For more context, see https://github.com/TriBITSPub/TriBITS/issues/522 and the commit log message for https://github.com/trilinos/Trilinos/pull/10930/commits/d03c5842af4a9d6bce3ef86ef10f2a596ef220cf.

If you take a step back, you can see why it has to be this way.  Suppose there is some old CMake project that ended active development 8 years ago that used Trilinos and then you want to build it was a newer version of Trilinos.  That project may have `cmake_minimum_required(VERSION 3.1)` or something in its top-level `CMakeLists.txt` file.  We would no longer be able to build that project against a modern install of Trilinos unless someone went in and manually modified that `CMakeLists.txt` file.   But why? That project has not changed at all.  Why should it have to bump up its minimum required CMake version?  By taking care of features used in our `<Package>Config.cmake` files ourselves, we can spare old downstream projects from having to change anything.

If you have more questions about this, let's have a short conversation to clear this up.